modules/networking/main.tf

```diff
@@ -1,129 +1,128 @@
-data "aws_availability_zones" "available" {}
-
 # Creates a VPC
 resource "aws_vpc" "vpc" {
-    cidr_block = var.cidr
+  cidr_block = var.cidr
 
-    tags = {
-        Name = "${var.app-name}-VPC"
-    }
+  tags = {
+    Name = "${var.app_name}-VPC"
+  }
 }
 
-# Creates 3 public subnets in the VPC"
+# Creates public subnets in the VPC"
 resource "aws_subnet" "public_subnets" {
-    count = length(var.public_subnets_list)
+  count = length(var.public_subnets_list)
 
-    vpc_id = var.id_of_vpc
-    map_public_ip_on_launch = true
-    cidr_block = var.public_subnets_list[count.index]
-    availability_zone = var.availability_zones[count.index]
+  vpc_id                  = aws_vpc.vpc.id
+  map_public_ip_on_launch = true
+  cidr_block              = var.public_subnets_list[count.index]
+  availability_zone       = var.availability_zones[count.index]
 
-    tags = {
-        Name = "${var.app-name}-Public-Subnet"
-    }
+  tags = {
+    Name = "${var.app_name}-Public-Subnet"
+  }
 }
 
-# Creates 3 private subnets in the VPC
+# Creates private subnets in the VPC
 resource "aws_subnet" "private_subnets" {
-    count = length(var.private_subnets_list)
+  count = length(var.private_subnets_list)
 
-    vpc_id = var.id_of_vpc
-    map_public_ip_on_launch = true
-    cidr_block = var.private_subnets_list[count.index]
-    availability_zone = var.availability_zones[count.index]
+  vpc_id                  = aws_vpc.vpc.id
+  map_public_ip_on_launch = true
+  cidr_block              = var.private_subnets_list[count.index]
+  availability_zone       = var.availability_zones[count.index]
 
-    tags = {
-        Name = "${var.app-name}-Private-Subnet"
-    }
+  tags = {
+    Name = "${var.app_name}-Private-Subnet"
+  }
 }
 
 # Creates DB Subnet group
 resource "aws_db_subnet_group" "db_subnets" {
-    name = "${var.app-name}-db-subnets"
+  count = var.enable_db_subnets ? 1 : 0
+  name  = "${var.app_name}-db-subnets"
 
-    subnet_ids = flatten([aws_subnet.private_subnets.*.id])
+  subnet_ids = flatten([aws_subnet.private_subnets.*.id])
 
-    tags = {
-        Name = "${var.app-name}-DB-Subnet-Group"
-    }
+  tags = {
+    Name = "${var.app_name}-DB-Subnet-Group"
+  }
 
 }
 
 # Creates an Internet Gateway in the VPC
 resource "aws_internet_gateway" "ig" {
-    vpc_id = var.id_of_vpc
+  vpc_id = aws_vpc.vpc.id
 
-    tags = {
-        Name = "${var.app-name}-Internet-Gateway"
-    }
+  tags = {
+    Name = "${var.app_name}-Internet-Gateway"
+  }
 }
 
 # Creates 3 NAT gateways in each of the public subnets
 resource "aws_nat_gateway" "nat" {
-  count = var.just_count
-  subnet_id = element(aws_subnet.public_subnets.*.id, count.index)
+  count         = var.just_count
```

Comment:
is there another way to set count value here?

Reply:
Yes, I researched and removed var.just_count and used length(var.public_subnets_list) instead for both NAT gateways, EIPs and private route tables.